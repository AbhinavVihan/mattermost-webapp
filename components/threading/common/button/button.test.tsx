// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {fireEvent, screen} from '@testing-library/react';

import ReplyIcon from 'components/widgets/icons/reply_icon';

import {renderWithIntl} from 'tests/react_testing_utils';

import Button from './button';

describe('components/threading/common/button', () => {
    test('should support onClick', () => {
        const action = jest.fn();

        const wrapper = renderWithIntl(
            <Button
                onClick={action}
            />,
        );

        expect(wrapper).toMatchSnapshot();

        fireEvent.click(screen.getByTestId('Button___transparent'));

        expect(action).toHaveBeenCalled();
    });

    test('should support className', () => {
        const className = 'test-class other-test-class';
        const wrapper = renderWithIntl(
            <Button
                className={className}
            />,
        );

        expect(wrapper).toMatchSnapshot();

        const button = screen.getByTestId('Button___transparent');
        expect(button).toHaveClass('test-class');
        expect(button).toHaveClass('other-test-class');
    });

    test('should support prepended content', () => {
        const wrapper = renderWithIntl(
            <Button
                prepend={<ReplyIcon className='Icon'/>}
            />,
        );

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.container.className.includes('Button_prepended ReplyIcon'));
    });

    test('should support appended content', () => {
        const wrapper = renderWithIntl(
            <Button
                append={<ReplyIcon className='Icon'/>}
            />,
        );

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.container.className.includes('Button_appended ReplyIcon'));
    });

    test('should support children', () => {
        const wrapper = renderWithIntl(
            <Button>
                {'text-goes-here'}
            </Button>,
        );

        expect(wrapper).toMatchSnapshot();
        const button = screen.getByTestId('Button___transparent');
        expect(button).toHaveTextContent('text-goes-here');
    });
});
