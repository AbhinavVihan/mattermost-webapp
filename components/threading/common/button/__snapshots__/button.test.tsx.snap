// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/threading/common/button should support appended content 1`] = `
<button
  className="mm-button mm-button__transparent"
>
  <span
    className="Button_label"
  />
  <span
    className="Button_appended"
  >
    <ReplyIcon
      className="Icon"
    />
  </span>
</button>
`;

exports[`components/threading/common/button should support children 1`] = `
<button
  className="mm-button mm-button__transparent"
>
  <span
    className="Button_label"
  >
    text-goes-here
  </span>
</button>
`;

exports[`components/threading/common/button should support className 1`] = `
<button
  className="mm-button mm-button__transparent test-class other-test-class"
>
  <span
    className="Button_label"
  />
</button>
`;

exports[`components/threading/common/button should support onClick 1`] = `
<button
  className="mm-button mm-button__transparent"
  onClick={[MockFunction]}
>
  <span
    className="Button_label"
  />
</button>
`;

exports[`components/threading/common/button should support prepended content 1`] = `
<button
  className="mm-button mm-button__transparent"
>
  <span
    className="Button_prepended"
  >
    <ReplyIcon
      className="Icon"
    />
  </span>
  <span
    className="Button_label"
  />
</button>
`;
